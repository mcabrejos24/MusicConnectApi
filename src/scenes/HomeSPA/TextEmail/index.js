import { useState } from "react";
import BackButton from "../../../components/BackButton";
import MusicCard2 from "../../../components/MusicCard2";
import AuthorizerAppleMusic from "../../../components/AuthorizerAppleMusic";
import AuthorizerSpotify from "../../../components/AuthorizerSpotify";
import Submit from "../../../components/Submit";

export default function TextEmail() {
  const [playlistReady, setPlaylistReady] = useState(false);
  const [contactReady, setContactReady] = useState(false);


  function openPopUp(pop) {
    const popup = document.getElementById("popup");
    popup.style.visibility = "visible";
    popup.style.opacity = "1";
  }

  function closePopUp() {
    const popup = document.getElementById("popup");
    popup.style.visibility = "hidden";
    popup.style.opacity = "0";
  }

  setTimeout(openPopUp,500);

  return (
    <div className="content page-text-email">

      <div className="coming-soon">
        <div id="popup" className="popup">
          <div className="popup-card">
            <h2>Coming Soon!</h2>
            <button id="close" class="close" onClick={closePopUp}>&times;</button>
            <p className="popup-card--content">This page is still in development so submitting won't actually do anything. But feel free to mess around!</p>
          </div>
        </div>
      </div>

      <BackButton></BackButton>
      <div className="double-authorizer-wrapper">
        <MusicCard2
          confirmPlaylist={setPlaylistReady}
          setContactReady={setContactReady}
        >
          <AuthorizerAppleMusic />
          <AuthorizerSpotify />
        </MusicCard2>
      </div>
      <Submit playlistReady={playlistReady} contactReady={contactReady} />

      <div className="how-it-works">
          <h1 className="how-it-works-title">How to use this sync card</h1>
          <div className="point need">
              <h2 className="point-title">Choose</h2>
              <div className="point-description">
                  <p>This one is for when you peer is not with you.
                      <br /><br />
                      Simply, you log in with youe Apple Music or Spotify account, whichever you use and want to sync.
                      <br /><br />
                      You then choose the playlist you would like to sync and provide your peers email or phone number, whichever method they prefer to receive a link to pair with you.
                      <br /><br />
                      If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
                      <br /><br />
                      Then just press the Submit button and we'll send a link to your peer, so that they can submit their information.
                      <br /><br />
                      Once your peer submits their information to sync, then your playlist will sync ahead and be good to go!.
                  </p>
              </div>
          </div>
      </div>
    </div>
  );
}
