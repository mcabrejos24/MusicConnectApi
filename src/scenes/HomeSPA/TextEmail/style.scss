.double-authorizer-wrapper {
    display: flex;
    padding: 86px 0 65px 0;
    justify-content: center;
}