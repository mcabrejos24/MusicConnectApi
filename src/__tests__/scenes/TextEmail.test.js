import React from "react";
import { shallow } from "enzyme";
import TextEmail from '../../scenes/HomeSPA/TextEmail';
import toJson from "enzyme-to-json";

describe("renders TextEmail page without useEffect hooke", () => {
    it("TextEmail renders", () => {
        shallow(<TextEmail/>);
    });
});

describe("renders TextEmail page", () => {
    let page;
    beforeEach(() => {
        page = shallow(<TextEmail isTest={true}/>);
        setTimeout(
            page.find(".test-element.hidden").find("#testOpenPopUp").simulate("click"),
            500
        );
    });

    it("TextEmail loads", () => {
        page;
    });
    it("props class name matches", () => {
        expect(page.props().className).toEqual("content page-text-email");
    });
    it("matches snapshot", () => {
        expect(toJson(page)).toMatchSnapshot();
    });
    it("pop up renders", () => {
        expect(page.find(".popup").text()).toEqual("Coming Soon!Ã—This page is still in development so submitting won't actually do anything. But feel free to mess around!");
    });
    it("pop up renders", () => {
        expect(page.find(".popup").hasClass("popup-show")).toEqual(true);
        expect(page.find(".popup").hasClass("popup-hide")).toEqual(false);
    });
    it("can dismiss popup", () => {
        page.find(".popup").find(".popup-card").find("#close").simulate("click");
        expect(page.find(".popup").hasClass("popup-show")).toEqual(false);
        expect(page.find(".popup").hasClass("popup-hide")).toEqual(true);
    });
});
