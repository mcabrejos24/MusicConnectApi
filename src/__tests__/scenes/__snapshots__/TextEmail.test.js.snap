// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TextEmail page can dismiss popup 1`] = `
<div
  className="content page-text-email"
>
  <div
    className="test-element hidden"
  >
    <button
      id="testOpenPopUp"
      onClick={[Function]}
    />
  </div>
  <div
    className="coming-soon"
  >
    <div
      className="popup popup-hide"
      id="popup"
    >
      <div
        className="popup-card"
      >
        <h2>
          Coming Soon!
        </h2>
        <button
          className="close"
          id="close"
          onClick={[Function]}
        >
          ×
        </button>
        <p
          className="popup-card--content"
        >
          This page is still in development so submitting won't actually do anything. But feel free to mess around!
        </p>
      </div>
    </div>
  </div>
  <BackButton />
  <div
    className="double-authorizer-wrapper"
  >
    <MusicCard2
      confirmPlaylist={[Function]}
      setContactReady={[Function]}
    >
      <AuthorizerAppleMusic />
      <AuthorizerSpotify />
    </MusicCard2>
  </div>
  <SyncElement
    contactReady={false}
    playlistReady={false}
  />
  <div
    className="how-it-works text-email"
  >
    <h1
      className="how-it-works-title"
    >
      How to use this sync card
    </h1>
    <div
      className="point need"
    >
      <h2
        className="point-title"
      >
        Choose
      </h2>
      <div
        className="point-description"
      >
        <p>
          This one is for when you peer is not with you.
          <br />
          <br />
          Simply, you log in with youe Apple Music or Spotify account, whichever you use and want to sync.
          <br />
          <br />
          You then choose the playlist you would like to sync and provide your peers email or phone number, whichever method they prefer to receive a link to pair with you.
          <br />
          <br />
          If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
          <br />
          <br />
          Then just press the Submit button and we'll send a link to your peer, so that they can submit their information.
          <br />
          <br />
          Once your peer submits their information to sync, then your playlist will sync ahead and be good to go!.
          <br />
          <br />
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders TextEmail page matches snapshot 1`] = `
<div
  className="content page-text-email"
>
  <div
    className="test-element hidden"
  >
    <button
      id="testOpenPopUp"
      onClick={[Function]}
    />
  </div>
  <div
    className="coming-soon"
  >
    <div
      className="popup popup-show"
      id="popup"
    >
      <div
        className="popup-card"
      >
        <h2>
          Coming Soon!
        </h2>
        <button
          className="close"
          id="close"
          onClick={[Function]}
        >
          ×
        </button>
        <p
          className="popup-card--content"
        >
          This page is still in development so submitting won't actually do anything. But feel free to mess around!
        </p>
      </div>
    </div>
  </div>
  <BackButton />
  <div
    className="double-authorizer-wrapper"
  >
    <MusicCard2
      confirmPlaylist={[Function]}
      setContactReady={[Function]}
    >
      <AuthorizerAppleMusic />
      <AuthorizerSpotify />
    </MusicCard2>
  </div>
  <SyncElement
    contactReady={false}
    playlistReady={false}
  />
  <div
    className="how-it-works text-email"
  >
    <h1
      className="how-it-works-title"
    >
      How to use this sync card
    </h1>
    <div
      className="point need"
    >
      <h2
        className="point-title"
      >
        Choose
      </h2>
      <div
        className="point-description"
      >
        <p>
          This one is for when you peer is not with you.
          <br />
          <br />
          Simply, you log in with youe Apple Music or Spotify account, whichever you use and want to sync.
          <br />
          <br />
          You then choose the playlist you would like to sync and provide your peers email or phone number, whichever method they prefer to receive a link to pair with you.
          <br />
          <br />
          If you want to make a new playlist, then enter the name of the new playlist and press the button below it to create it.
          <br />
          <br />
          Then just press the Submit button and we'll send a link to your peer, so that they can submit their information.
          <br />
          <br />
          Once your peer submits their information to sync, then your playlist will sync ahead and be good to go!.
          <br />
          <br />
        </p>
      </div>
    </div>
  </div>
</div>
`;
